MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/KotlinStyleCirclesCategoricallyActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.mapbox.mapboxandroiddemo.examples.dds
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.mapbox.mapboxandroiddemo.R
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.maps.MapView
+import com.mapbox.mapboxsdk.style.expressions.Expression.*
+import com.mapbox.mapboxsdk.style.layers.CircleLayer
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleColor
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory.circleRadius
+import com.mapbox.mapboxsdk.style.sources.VectorSource
+
+/**
+ * Kotlin example of using data-driven styling to set circles' colors based on imported vector data.
+ */
+class KotlinStyleCirclesCategoricallyActivity : AppCompatActivity() {
+    private lateinit var mapView: MapView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Mapbox access token is configured here. This needs to be called either in your application
+        // object or in the same activity which contains the mapview.
+        Mapbox.getInstance(this, getString(R.string.access_token))
+
+        // This contains the MapView in XML and needs to be called after the access token is configured.
+        setContentView(R.layout.activity_dds_kotlin_style_circles_categorically)
+
+        mapView = findViewById(R.id.mapView)
```

Comment:
with kotlin android extensions functions you can remove the findViewById call and the lateinit MapView (requires ktx dep). This should be fixed on the simple map activity example as well.